projects/angular/src/utils/focus-trap/focus-trap.ts

```diff
@@ -0,0 +1,189 @@
+/*
```

Comment:
This is a copy of the focus trap we have in Clarity Core that is about to be deprecated. It works better than the `[clrFocusTrap]` so we might take a closer look down the road to reuse this rather than `[clrFocusTrap]`

Reply:
That was the first thing I tried but the problem with it is that the [CdsBaseFocusTrap](https://github.com/vmware-clarity/core/blob/1470155a9581e31020db4e9c0397534142fabfb8/packages/core/src/internal/base/focus-trap.base.ts#L14) extends `LitElement` and that causes compile issues in angular when you try for example to do `ClrNavLevel extends CdsBaseFocusTrap`

Reply:
Yes, but what we have here is a reverse engineer of the previous focus trap. I think it would make more sense to reverse engineer the new one...

Reply:
I believe we can do that but I would prefer to get into this when the v6 is released. What do you think of that idea?

Reply:
It's pushing more work (a.k.a. tech debt) out into the future and may be harder (more expensive) to make these changes then. It's also accepting a known less-performant solution in the short term. Which may come back to bite us.

What is your level of confidence that this tech debt will be addressed? And when?

Reply:
When this one is merged and v6 is released I'll be spending time getting the new focus trap functionality plugged into the library. 

The reason for this merging it like this at this stage is to not postpone the critical bugfix anymore.